src/stcal/ramp_fitting/ramp_fit.py

```diff
@@ -0,0 +1,104 @@
+#! /usr/bin/env python
+#
+#  ramp_fit.py - calculate weighted mean of slope, based on Massimo
+#                Robberto's "On the Optimal Strategy to fit MULTIACCUM
+#                ramps in the presence of cosmic rays."
+#                (JWST-STScI-0001490,SM-12; 07/25/08).   The derivation
+#                is a generalization for >1 cosmic rays, calculating
+#                the slope and variance of the slope for each section
+#                of the ramp (in between cosmic rays). The intervals are
+#                determined from the input data quality arrays.
+#
+# Note:
+# In this module, comments on the 'first group','second group', etc are
+#    1-based, unless noted otherwise.
+
+import numpy as np
+import logging
+
+# from . import gls_fit           # used only if algorithm is "GLS"
+from . import ols_fit           # used only if algorithm is "OLS"
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+BUFSIZE = 1024 * 300000  # 300Mb cache size for data section
+
+
+def ramp_fit(model, buffsize, save_opt, readnoise_2d, gain_2d,
+             algorithm, weighting, max_cores):
```

Comment:
This will have to be discussed what the best way is to get this information to ramp fit in STCAL.  Changing the `ramp_fit` function will mean changing all the tests, but that might be the best way to make this function general use for multiple pipelines.